src/oss/langchain-messages.mdx

```diff
@@ -2,14 +2,488 @@
+response = model.invoke(human_msg)
+response = model.invoke("What is machine learning?")
+```
+
+### String Shortcut
+
+TODO keep??
+
+For quick testing, strings are automatically converted to HumanMessage:
+
+```python
+# These are equivalent
+response1 = model.invoke([HumanMessage("Hello")])
+response2 = model.invoke("Hello")  # Automatically converted
+```
+
+### Message Metadata
+
+```python
+# Add metadata
+human_msg = HumanMessage(
+    content="Hello!",
+    name="alice",  # Identify different users
+    id="msg_123"   # Unique identifier for tracing
+)
+
+# Use in multi-user scenarios
+messages = [
+    HumanMessage("I like apples", name="alice"),
+    HumanMessage("I prefer oranges", name="bob"),
+    HumanMessage("What fruits do Alice and Bob like?", name="charlie")
+]
+```
+
+TODO make sure how `name` field is used across different providers is clear
+
 ## AI Message
 
+AI messages represent model responses and contain the output of chat model invocations. They can include text, tool calls, and rich metadata.
+
+### Basic AI Messages
+
+```python
+from langchain_core.messages import AIMessage
+
+# Simple response
+ai_msg = AIMessage("Machine learning is a subset of artificial intelligence...")
+
+# Access response content
+response = model.invoke("Explain AI")
+```
+
+### AI Message Attributes
+
+```python
+response = model.invoke("Tell me a joke")
+
+# Standardized attributes
+print(f"Content: {response.content}")
+print(f"Tool calls: {response.tool_calls}")
+print(f"Usage: {response.usage_metadata}")
+print(f"ID: {response.id}")
+
+# Provider-specific response metadata
+print(f"Response metadata: {response.response_metadata}")
+```
+
+### Response Metadata
+
+AI messages include metadata from the provider with token usage, model details, and generation information:
+
+```python
+response = model.invoke("Explain quantum computing")
+metadata = response.response_metadata
+
+print(f"Model: {metadata.get('model_name')}")
+print(f"Tokens: {metadata.get('token_usage', {}).get('total_tokens', 0)}")
+print(f"Finish reason: {metadata.get('finish_reason')}")
+```
+
+Common fields include `model_name`, `finish_reason`, `id`, and `token_usage` (format varies by provider).
+
+### Tool Calling Responses
+
+When models make tool calls, they're included in the AI message:
+
+TODO schema switcher
+
+```python
+from pydantic import BaseModel, Field
+
+class GetWeather(BaseModel):
+    """Get weather for a location"""
+    location: str = Field(..., description="City and state")
+
+llm_with_tools = llm.bind_tools([GetWeather])
+response = llm_with_tools.invoke([HumanMessage("What's the weather in Paris?")])
+
+# Access tool calls
+if response.tool_calls:
+    for tool_call in response.tool_calls:
+        print(f"Tool: {tool_call['name']}")
+        print(f"Args: {tool_call['args']}")
+        print(f"ID: {tool_call['id']}")
+```
+
+### Streaming and Chunks
+
+During streaming, you receive `AIMessageChunk` objects that accumulate metadata:
```

Comment:
i would just call this `AI MessageChunk`

what are properties of AI message chunk